src/app/fdctl/run/tiles/fd_shred.c

```diff
@@ -700,11 +701,13 @@ unprivileged_init( fd_topo_t *      topo,
 
   ulong expected_shred_version = tile->shred.expected_shred_version;
   if( FD_LIKELY( !expected_shred_version ) ) {
-    ulong * gossip_shred_version = (ulong*)tile->extra[0];
+    ulong busy_obj_id = fd_pod_query_ulong( topo->props, "poh_shred", ULONG_MAX );
+    FD_TEST( busy_obj_id!=ULONG_MAX );
+    ulong * gossip_shred_version = fd_fseq_join( (void*)((ulong)(topo->workspaces[ topo->objs[ busy_obj_id ].wksp_id ].wksp) + topo->objs[ busy_obj_id ].offset) );
     FD_LOG_INFO(( "Waiting for shred version to be determined via gossip." ));
-    while( !expected_shred_version ) {
+    do {
```

Comment:
Shred version of 0 is an invalid shred version used for "unknown" or "not yet assigned," but shred version is a `ushort`, so I guess `ULONG_MAX` is fine as a sentinel.

Reply:
This changed just because we fd_fseq_new with ULONG_MAX, and I didn't want to special case it.